src/main/java/org/fcrepo/migration/idmappers/OpaqueIDMapper.java

```diff
@@ -0,0 +1,180 @@
+package org.fcrepo.migration.idmappers;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.StringField;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.SearcherManager;
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FSDirectory;
+import org.apache.lucene.util.Version;
+import org.fcrepo.migration.Fedora4Client;
+import org.fcrepo.migration.MigrationIDMapper;
+import org.slf4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * An IDMapper implementation that maps identifiers by simply creating new
+ * objects in the Fedora 4 instance and thus delegating to whatever pid minter
+ * is configured for the repository.  Unlike other ID mappers, just getting an
+ * ID through mapObjectPath creates an object in the repository!
+ *
+ * Meanwhile, it maintains a persistent copy of the mapping in a Lucene Index
+ * on the filesystem.
+ *
+ * @author Mike Durbin
+ */
+public class OpaqueIDMapper implements MigrationIDMapper {
+
+    private static final Logger ID_LOGGER = getLogger("id-mapper");
+    private static final Logger LOGGER = getLogger(OpaqueIDMapper.class);
+
+    private Fedora4Client f4Client;
+
+    /**
+     * A lucene SearcherManager over an index maintained by this class.
+     * For object created as part of this migration operation a document
+     * exists in this index that contains a "pid" field and a "path" field.
+     * The pid field is the original fedora 3 pid, the path field is the
+     * path within the destination repository for that pid.
+     */
+    private SearcherManager searcherManager;
+
+    /**
+     * A lucene IndexWriter for the index exposed by 'searcher'.
+     */
+    private IndexWriter writer;
+
+    /**
+     * A constructor.
+     * @param cachedIDIndexDir the directory (or null) where the index of generated pids should be maintained
+     * @param f4Client a Fedora 4 client to mediate interactions with the repository
+     * @throws IOException
+     */
+    public OpaqueIDMapper(final File cachedIDIndexDir, final Fedora4Client f4Client) throws IOException {
+        this.f4Client = f4Client;
+        final File indexDir;
+        if (cachedIDIndexDir == null) {
+            final File temp = File.createTempFile("tempfile", "basedir");
+            temp.delete();
+            temp.mkdir();
+            indexDir = new File(temp, "index");
+            LOGGER.info("No generated ID index directory specified.  Creating temporary index at \""
```

Comment:
Not critical, but you should probably be using the more efficient slf4j logging syntax:
`mporary index at \"{}\".", indexDir.getAbsolutePath());`